compiler/passes/errorHandling.cpp

```diff
@@ -603,3 +743,59 @@ bool isCheckErrorStmt(Expr* e)
   }
   return false;
 }
+
+// Should we raise an error in strict mode if the error is not handled?
+// No for calls inside of compiler-generated functions, wrapper functions,
+// or task functions. No for functions marked with FLAG_UNCHECKED_THROWS.
+static bool isCompilerGeneratedFunction(FnSymbol* fn)
+{
+  return isTaskFun(fn) ||
+         fn->hasFlag(FLAG_WRAPPER) ||
+         fn->hasFlag(FLAG_COMPILER_GENERATED);
+}
+
+static bool isUncheckedThrowsFunction(FnSymbol* fn)
+{
+  return fn->hasFlag(FLAG_UNCHECKED_THROWS);
+}
+
+
+void lowerErrorHandling() {
+  if (!fMinimalModules)
+    INT_ASSERT(dtError->inTree());
+
+  std::set<FnSymbol*> visited;
+  implicitThrowsReasons_t reasons;
+
+  forv_Vec(FnSymbol, fn, gFnSymbols) {
+    // Determine if compiler-generated fns should be marked 'throws'
+    markImplicitThrows(fn, &visited, &reasons);
+  }
+
+  forv_Vec(FnSymbol, fn, gFnSymbols) {
+    // Otherwise, just check for error-handling errors.
+    checkErrorHandling(fn, &reasons);
+  }
+
+  // Quit if fatal errors were encountered by checkErrorHandling above.
+  USR_STOP();
+
+  forv_Vec(FnSymbol, fn, gFnSymbols) {
+    ArgSymbol*   outError = NULL;
+    LabelSymbol* epilogue = NULL;
+
+    if (fn->throwsError()) {
+      SET_LINENO(fn);
+
+      outError = new ArgSymbol(INTENT_REF, "error_out", dtError);
+      outError->addFlag(FLAG_ERROR_VARIABLE);
+      fn->insertFormalAtTail(outError);
+
+      epilogue = fn->getOrCreateEpilogueLabel();
+      INT_ASSERT(epilogue); // throws requires an epilogue
+    }
+
+    ErrorHandlingVisitor visitor = ErrorHandlingVisitor(outError, epilogue);
+    fn->accept(&visitor);
```

Comment:
I suppose by my `checkImplicitThrows` logic, lines 784 to 799 should be made into a static method of `ErrorHandlingVisitor`. I'm not sure that'd be best, but it'd make `lowerErrorHandling` aesthetically pleasing with three single-statement loops.